File: CSharp/ImageProcessing/Lib/ComInfo.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ImageProcessing
{
    public class ComInfo
    {
        public enum Pixel
        {
            B = 0,
            G,
            R,
            A,
            MAX,
        };

        public enum ImgType
        {
            EdgeDetection = 0,
            MAX,
        }

        public enum ImgDataType
        {
            Jpg = 0,
            Png,
            MAX,
        }

        public const string MENU_SETTING_IMAGE_PROCESSING = "Image Processing";
        public const string MENU_FILE_END = "End(_X)";
        public const string MENU_SAVE_CSV = "Save Csv...";
        public const string BTN_OK = "btnOk";
        public const string BTN_CANCEL = "btnCancel";
        public const string IMG_NAME_EDGE_DETECTION = "EdgeDetection";
        public const string IMG_NAME_GRAY_SCALE = "GrayScale";
        public const string IMG_NAME_BINARIZATION = "Binarization";
        public const string IMG_NAME_GRAY_SCALE_2DIFF = "GrayScale 2Diff";
        public const string IMG_NAME_COLOR_REVERSAL = "ColorReversal";
    }
}

File: Vb/ImageProcessing/Lib/ComInfo.vb
Imports System.Runtime.InteropServices.Marshal
Imports System.Text
Imports LiveCharts

''' <summary>
''' 共通情報のロジック
''' </summary>
Public Class ComInfo
    ''' <summary>
    ''' Pixelの列挙
    ''' </summary>
    Public Enum Pixel
        B = 0
        G
        R
        A
    End Enum

    ''' <summary>
    ''' 画像処理のタイプの列挙
    ''' </summary>
    Public Enum ImageProcessingType
        EdgeDetection = 0
        GrayScale
        Binarization
        GrayScale2Diff
        ColorReversal
        MAX
    End Enum

    ''' <summary>
    ''' 画像のタイプの列挙
    ''' </summary>
    Public Enum ImageDataType
        Jpg = 0
        Png
        MAX
    End Enum

    ''' <summary>
    ''' 画像の状態の列挙
    ''' </summary>
    Public Enum PictureType
        Original = 0
        After
        MAX
    End Enum

    Public Const RGB_MAX As Short = 256

    Public Const MENU_SETTING_IMAGE_PROCESSING As String = "Image Processing"
    Public Const MENU_FILE_END As String = "End(_X)"
    Public Const MENU_SAVE_FILE As String = "Save Csv..."
    Public Const BTN_OK As String = "btnOk"
    Public Const BTN_CANCEL As String = "btnCancel"
    Public Const IMG_NAME_EDGE_DETECTION As String = "EdgeDetection"
    Public Const IMG_NAME_GRAY_SCALE As String = "GrayScale"
    Public Const IMG_NAME_BINARIZATION As String = "Binarization"
    Public Const IMG_NAME_GRAY_SCALE_2DIFF As String = "GrayScale 2Diff"
    Public Const IMG_NAME_COLOR_REVERSAL = "ColorReversal"
End Class